### PR TITLE
feat: add workflows and checks

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

Adds workflows as polled in https://canary.discord.com/channels/881118111967883295/962685290294673478

also removes CODEOWNERS as it only works for maintainers

## NOTE
this targets a feature branch so all issues (pyright, format) can be resolved before adding to master

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -7,7 +7,8 @@
 <!-- Put an x inside [ ] to check it, like so: [x] -->
 
 - [ ] If code changes were made then they have been tested.
-    - [ ] I have updated the documentation to reflect the changes.
+  - [ ] I have updated the documentation to reflect the changes.
+  - [ ] I ran `task pyright` and fixed the relevant issues.
```
Comment (MaskDuck) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
  - [ ] I have ran `task pyright` and fixed the relevant issues.
```

**Review by MaskDuck — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,7 +7,8 @@
 <!-- Put an x inside [ ] to check it, like so: [x] -->
 
 - [ ] If code changes were made then they have been tested.
-    - [ ] I have updated the documentation to reflect the changes.
+  - [ ] I have updated the documentation to reflect the changes.
+  - [ ] I have ran `task pyright` and fixed the relevant issues.
```
Comment (DenverCoder1) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: This should be either "I ran" or "I have run". The past participle of "ran" is "run".

```suggestion
  - [ ] I have run `task pyright` and fixed the relevant issues.
```

Comment (ooliver1) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: it was from https://github.com/nextcord/nextcord/pull/596#discussion_r848267473 sorry

**Review by Skelmis — APPROVED** [review]

**Comment by Skelmis** [comment]
Approved, although pyright may be a tad much given it's currently failing right? Unless you aim to also fix that

**Comment by ooliver1** [comment]
> Approved, although pyright may be a tad much given it's currently failing right? Unless you aim to also fix that

Pyright basic with a few custom settings, this targets a branch so all fails can be resolved before in master

**Comment by MaskDuck** [comment]
kinda confused will pyright ever become required?

**Comment by ooliver1** [comment]
> kinda confused will pyright ever become required?

when its in master?

**Comment by ooliver1** [comment]
pre-commit.ci run

**Comment by ooliver1** [comment]
closing and opening to see if thats how pre-commit.ci works

**Comment by ooliver1** [comment]
commit ci may only run when its a pr to master, will see when that comes

**Pull request merged**